Tests/Client/Unit Tests/WebSocket/WebSocketTests.swift

```diff
@@ -118,6 +113,47 @@ final class WebSocketTests: XCTestCase {
         time.run(numberOfSeconds: 3 * pingInterval)
         XCTAssertEqual(socketProvider.sendPingCalledCounter, 1 + 3)
     }
+    
+    func test_typingEvent_stopTyping_afterTimeout() {
+        // Setup (connect)
+        test_connectionFlow()
+        
+        // Set `shouldAutomaticallySendTypingStopEvent` to `true`
+        let delegate = MockDelegate()
+        delegate.shouldAutomaticallySendTypingStopEvent = true
+        webSocket.eventDelegate = delegate
+        
+        // Simulate some user started typing
+        let otherUser = User(id: UUID().uuidString)
+        socketProvider.simulateMessageReceived(.typingStartEvent(userId: otherUser.id))
+        
+        var nextEvent: Event?
+        _ = webSocket.subscribe { nextEvent = $0 }
```

Comment:
@VojtaStavik do you subscribed here when the event already was sent?

Reply:
It doesn't matter, we use async assert for that, so as long as `nextEvent` will eventually be `typingStop`, the test works.